framework/src/variables/MooseVariableBase.C

```diff
@@ -189,13 +190,16 @@ MooseVariableBase::componentDofIndices(const std::vector<dof_id_type> & dof_indi
 void
 MooseVariableBase::scalingFactor(Real factor)
 {
-  _scaling_factor.assign(_count, factor);
+  for (auto & v : _scaling_factor)
+    v = factor;
 }
```

Comment:
The coverage report would tell us this but we didn't get it because of the THM threads failure. I'm wondering whether we can just remove this overload

Reply:
Will try and report back. This is a legacy.

Reply:
Looks like we can. The only place this function is used is in `SystemBase::applyScalingFactors` actually.